Source/LDSwiftEventSource.swift

```diff
@@ -288,14 +288,19 @@ class EventSourceDelegate: NSObject, URLSessionDataDelegate {
 
         // swiftlint:disable:next force_cast
         let httpResponse = response as! HTTPURLResponse
-        if (200..<300).contains(httpResponse.statusCode) {
+        let statusCode = httpResponse.statusCode
+        if statusCode == 204 {
+            logger.log(.info, "Server requesting no further retries: %d", statusCode)
+            readyState = .shutdown
```

Comment:
Ok, for the sake of completing the context that I may be missing.

If the readyState becomes shutdown, either via this 204, or via the `connectionErrorHandler`, then the events source is done at that point, and presumably the SDK will not attempt to make a new one and reconnect.

The fact that the event source is stopping, because it got a 204, wouldn't ever be communicated to a consumer of the event source? (it may be communicated, and I am just unfamiliar with the code.)

This also may be fine, I am just thinking about what the SDK behavior may be versus the event source.

Reply:
Discussed internally. 204 is going to be treated like a failure condition, like the JS event source would. This allows us to provide a default behavior where 204 shuts it down, but allows consumers to override that behavior if they wish.